test/cpp/lazy/test_ir.cpp

```diff
@@ -97,5 +100,28 @@ TEST(IrTest, TsNode) {
   EXPECT_TRUE(leafptr != nullptr);
 }
 
+TEST(IrTest, DimensionNodeTest) {
+
+  const size_t DIM0 = 5;
+  const size_t DIM1 = 8;
+  NodePtr node1 = MakeNode<TsNode>(
+      OpKind(at::aten::view),
+      Shape(c10::kFloat, {DIM0, DIM1}),
+      /*num_outputs*/ 1,
+      /*hash_seed*/ kHashSeed);
+
+  auto size0 = std::dynamic_pointer_cast<SizeNode>(MakeNode<SizeNode>(node1, 0));
+  auto size1 = std::dynamic_pointer_cast<SizeNode>(MakeNode<SizeNode>(node1, 1));
+
+  ASSERT_EQ(DIM0, size0->getStaticValue());
+  ASSERT_EQ(DIM1, size0->getStaticValue());
+
+  auto add_dim = std::dynamic_pointer_cast<SizeAdd>(MakeNode<SizeAdd>(size0, size1));
```

Comment:
how does this work? It looks like you compare equality of a size_t and a SizeAdd*.  I am looking for some method that performs implicit conversion of SizeAdd* into size_t, but not finding one.

Reply:
oops looks like you just fixed that. no further questions